src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+      }
+    }
+    scn.close();
+    return res;
+  }
+
+  /**
+   * Write a compilation unit to a Java file. The file that is written to is the file path obtained
+   * by appending the suffix to the end of the file's name, before the file type suffix.
+   *
+   * @param compUnit the compilation unit to write to file
+   * @param filePath the absolute path to the input Java file
+   * @param suffix the suffix to append to the name of the new Java file
+   * @return {@code String} representing the absolute path to the newly written Java file. Returns
+   *     {@code null} if error occurred in writing to the new file
+   */
+  private static String writeToFile(CompilationUnit compUnit, String filePath, String suffix) {
+    // Rename the overall class to [old class name][suffix]
+    new ClassRenamer().visit(compUnit, new String[] {getClassName(filePath), suffix});
+
+    // Create a new string to represent the new file name
+    String newFileName =
+        new StringBuilder(filePath).insert(filePath.lastIndexOf('.'), suffix).toString();
+
+    // Write the compilation unit to the new file
+    try (BufferedWriter bw = new BufferedWriter(new FileWriter(newFileName))) {
+      bw.write(compUnit.toString());
+    } catch (IOException e) {
+      System.err.println("Error writing to file: " + newFileName);
+      e.printStackTrace();
+      System.exit(1);
+    }
+
+    return newFileName;
+  }
+
+  /**
+   * Given the absolute file path to a Java file, return the simple class name of the file.
+   *
+   * @param filePath absolute path to a Java file
+   * @return the simple class name
+   */
+  public static String getClassName(String filePath) {
+    if (filePath == null) {
+      return null;
+    }
+
+    // Get the name of the file without full path
+    String fileName = Paths.get(filePath).getFileName().toString();
+    // Remove .java extension from file name
+    return fileName.substring(0, fileName.indexOf('.'));
+  }
+
+  /**
+   * Rename the overall class. Visit every class and interface declaration, of which we expect there
+   * to be one in the input test suite. The overall class will be renamed to class name + suffix.
+   */
+  private static class ClassRenamer extends VoidVisitorAdapter<Object> {
+    /**
+     * @param arg a String array where the first element is the class name and the second element is
+     *     the suffix that we will append
+     */
+    @Override
+    public void visit(ClassOrInterfaceDeclaration n, Object arg) {
+      String[] params = (String[]) arg;
+      String className = params[0];
+      String suffix = params[1];
+      if (className.equals(n.getName())) {
+        n.setName(className + suffix);
+      }
+    }
+  }
+
+  private static class TypeVisitor extends VoidVisitorAdapter<Object> {
+    /** @param arg a set of {@code Type} objects */
+    @SuppressWarnings("unchecked")
+    @Override
+    public void visit(ClassOrInterfaceType n, Object arg) {
+      Set<Type> params = (Set<Type>) arg;
```

Comment:
This is supposed to add onto a set containing all the types that appear within the Java file. So if the type `org.apache.commons.lang3.MutablePair` appears in the program, it will be added to the set of types. This is then used for type name substitutions to simplify `org.apache.commons.lang3.MutablePair` into `MutablePair` after adding an import statement.